rbd/locks_test.go

```diff
@@ -0,0 +1,124 @@
+//go:build !nautilus && ceph_preview
+// +build !nautilus,ceph_preview
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLocking(t *testing.T) {
+	conn := radosConnect(t)
+	require.NotNil(t, conn)
+	defer conn.Shutdown()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	require.NoError(t, err)
+	defer conn.DeletePool(poolname)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx.Destroy()
+
+	// Connect another cluster
+	conn1 := radosConnect(t)
+	require.NotNil(t, conn1)
+	defer conn1.Shutdown()
+
+	ioctx1, err := conn1.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx1.Destroy()
+
+	options := NewRbdImageOptions()
+	defer options.Destroy()
+	assert.NoError(t, options.SetUint64(ImageOptionOrder, uint64(testImageOrder)))
+
+	name := GetUUID()
+	err = CreateImage(ioctx, name, testImageSize, options)
+	assert.NoError(t, err)
+
+	t.Run("acquireLock", func(t *testing.T) {
```

Comment:
Can we have few additional API calls to more meaningfully test Exclusive and Shared locks? For example two different lock owners sharing a lock on an image, attempt to acquire an Exclusive lock by a different owner etc ?

Reply:
Added some more tests for Exclusive lock, PTAL.